### PR TITLE
GPC-183 Add indexes

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/resources/db/migration/V2__indexes.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    on acquirer_subscription_enrichment_field (acquirer_subscription_id);
+
+create index ed_acquirer_subscription_id_index
+    on event_data (acquirer_subscription_id);
+
+create index ed_deleted_at_index
+    on event_data (deleted_at);
+
+create index ed_polling_index
+    on event_data (when_created, acquirer_subscription_id, deleted_at);
```
Comment (ethanmills) [inline src/main/resources/db/migration/V2__indexes.sql:20]: I'm never quite sure when partial indexes are worth it, but it seems like here might be a good place? For `deleted_at IS NULL`?

**Review by ethanmills — APPROVED** [review]
Not sure what's going on with the test failure - looks like my test was accidentally relying on some implementation detail. Happy to take a look if you like

**Comment by ethanmills** [comment]
> Not sure what's going on with the test failure - looks like my test was accidentally relying on some implementation detail. Happy to take a look if you like

```kotlin
  private fun thereAreFiveEvents() {
    val acquirerSubscription = acquirerSubscriptionRepository.findAllByOauthClientIdAndEventTypeIsIn(
      DWP_EVENT_RECEIVER,
      listOf(EventType.DEATH_NOTIFICATION),
    ).first()
    listOf("1", "2", "3", "4", "5").map { sourceId ->

      val eventData = EventData(
        UUID.randomUUID(),
        acquirerSubscription.acquirerSubscriptionId,
        sourceId,
        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
      )
      eventDataRepository.save(eventData)
    }
  }
```
in EventQueryTest should make it deterministic

**Comment by Mjwillis** [comment]
> > Not sure what's going on with the test failure - looks like my test was accidentally relying on some implementation detail. Happy to take a look if you like
> 
> ```diff
> @@ -112,8 +112,8 @@
>          UUID.randomUUID(),
>          acquirerSubscription.acquirerSubscriptionId,
>          sourceId,
> -        LocalDateTime.of(2000, 5, 3, 12, 4, 3),
> -        LocalDateTime.of(2000, 5, 3, 12, 4, 3),
> +        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
> +        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
>        )
>        eventDataRepository.save(eventData)
>      }
> ```
> 
> in EventQueryTest should make it deterministic



> > Not sure what's going on with the test failure - looks like my test was accidentally relying on some implementation detail. Happy to take a look if you like
> 
> ```diff
> @@ -112,8 +112,8 @@
>          UUID.randomUUID(),
>          acquirerSubscription.acquirerSubscriptionId,
>          sourceId,
> -        LocalDateTime.of(2000, 5, 3, 12, 4, 3),
> -        LocalDateTime.of(2000, 5, 3, 12, 4, 3),
> +        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
> +        LocalDateTime.of(2000, 5, 3, 12, 4, sourceId.toInt()),
>        )
>        eventDataRepository.save(eventData)
>      }
> ```
> 
> in EventQueryTest should make it deterministic

Yes did this !

**Comment by olevett** [comment]
This MR appears to have failed because of a lock on terraform state

**Pull request merged**